test/adapters/UniswapAdapter.js

```diff
@@ -0,0 +1,473 @@
+            BigNumber.from("1000000000000000000000000000000000"),
+            { from: user }
+          );
+
+          const expDigg = await this.adapter.expectedDiggOut(
+            this.wbtcAmt,
+            this.exchangeName,
+            {
+              from: user,
+            }
+          );
+          const promise = this.adapter.buyLp(
+            "0x2260fac5e5542a773aa44fbcfedf7c193bc2c599",
+            this.wbtcAmt,
+            this.exchangeName,
+            0,
+            expDigg - 1,
+            {
+              from: user,
+            }
+          );
+          //console.log('checking state after purchase');
+          await checkState(this.exchangeName, this.adapter.address, true);
+        });
+      });
+    });
+  }
+  async function checkState(exchangeName, adapterAddress, isAfter) {
+    const [, , userSigner, recipientSigner] = await ethers.getSigners();
+    user = userSigner.address;
+    recipient = recipientSigner.address;
+    if (exchangeName == "UNISWAP") {
+      lpAddress = UNISWAP_LP;
+    }
+    if (exchangeName == "SUSHISWAP") {
+      lpAddress = SUSHISWAP_LP;
+    }
+    const wethArtifact = await artifacts.readArtifact("IWETH");
+    this.weth = new ethers.Contract(
+      "0xc02aaa39b223fe8d0a0e5c4f27ead9083c756cc2",
+      wethArtifact.abi,
+      ethers.provider
+    );
+    this.weth = this.weth.connect(userSigner);
+
+    const tokenArtifact = await artifacts.readArtifact("ERC20");
+    this.wbtc = new ethers.Contract(
+      "0x2260fac5e5542a773aa44fbcfedf7c193bc2c599",
+      tokenArtifact.abi,
+      ethers.provider
+    );
+    this.wbtc = this.wbtc.connect(userSigner);
+
+    this.digg = new ethers.Contract(
+      "0x798d1be841a82a273720ce31c822c61a67a601c3",
+      tokenArtifact.abi,
+      ethers.provider
+    );
+    this.digg = this.digg.connect(userSigner);
+
+    const uniswapArtifact = await artifacts.readArtifact("IUniswapV2Pair");
+    this.uniswapLp = new ethers.Contract(
+      lpAddress,
+      uniswapArtifact.abi,
+      ethers.provider
+    );
+    this.uniswapLp = this.uniswapLp.connect(userSigner);
+
+    const ethBalance = await provider.getBalance(user);
+    //console.log(`eth balance of user is ${ethBalance}`);
+    const ethBalanceAdapter = await provider.getBalance(adapterAddress);
+    //console.log(`eth balance of adapter is ${ethBalanceAdapter}`);
+    const wbtcBalanceAdapter = await this.wbtc.balanceOf(adapterAddress);
+    //console.log(`wbtc balance of adapter is ${wbtcBalanceAdapter}`);
+    const diggBalanceAdapter = await this.digg.balanceOf(adapterAddress);
+    //console.log(`digg balance of adapter is ${diggBalanceAdapter}`);
+    const lpBalance = await this.uniswapLp.balanceOf(user);
+    //console.log(`lp balance of user is ${lpBalance}`);
+    if (isAfter) {
+      assert.isTrue(diggBalanceAdapter < 3);
+      assert.isTrue(wbtcBalanceAdapter < 3);
+      assert.isTrue(lpBalance > 0);
+      assert.equal(ethBalanceAdapter, 0);

```

Comment:
We should test that it ends up with the exact numbers, instead of just checking `> 0` so we're not missing anything